client/src/components/Profile/UserInfo.js

```diff
@@ -15,82 +17,137 @@ const useStyles = makeStyles({
+        height: 54,
+        marginLeft: '1.2em',
+        marginTop: '1em',
+    },
+    container: {
+        marginLeft: '3em',
+    },
+}));
 
-const UserInfo = props => {
-    const [firstName, setFirstName] = useState('Ashly');
-    const [lastName, setLastName] = useState('Sanford');
-    const [email, setEmail] = useState('usermail@gmail.com');
-    const [location, setLocation] = useState('+1 (202) 555 0192');
+const UserInfo = () => {
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [email, setEmail] = useState('');
+    const [location, setLocation] = useState('');
     const [phone, setPhone] = useState('');
 
     const classes = useStyles();
+    const { profile } = useGlobalContext();
+    console.log(profile);
+
+    const updateProfileInfo = () => {
+        const userInfo = {
+            firstName,
+            lastName,
+            email,
+            location,
+            phone,
+        };
+        axios.put(`/profile/${profile.user._id}`, userInfo);
+        console.log('update changes clicked');
+    };
 
     return (
-        //  User Info Grid Parent Container
-        <Grid container item sm={12} className={classes.profileGrid}>
-            <Grid item sm={12}>
-                <Typography variant="h1" className={classes.headerStyles}>
+        <Grid container direction="column" className={classes.container}>
+            <Grid
+                item
+                container
+                alignContent="center"
+                sm={2}
+                className={classes.headerContainer}
+            >
+                <Typography variant="h1" className={classes.header}>
                     Profile
                 </Typography>
             </Grid>
-            {/* First Name  */}
-            <Grid item>
-                <InputLabel className={classes.labelStyles}>First Name</InputLabel>{' '}
-                <TextField
-                    variant="outlined"
-                    hinttext="Change First Name.."
-                    onChange={e => setFirstName(e.target.value)}
-                    defaultValue={firstName}
-                    className={classes.inputStyles}
-                />
-            </Grid>
+            <form className={classes.root} noValidate autoComplete="off">
+                <Grid
+                    item
+                    container
+                    sm={10}
+                    alignContent="flex-start"
+                    justify="flex-start"
+                    spacing={4}
+                >
+                    <Grid item>
+                        <InputLabel className={classes.labelStyles}>
+                            First Name
+                        </InputLabel>
+                        <TextField
+                            id="outlined"
+                            hinttext="Change first name.."
+                            variant="outlined"
+                            value={profile.firstName}
```

Comment:
I don't think this is going to work. The value of profile.firstName never changes when editing, so I believe the TextFields will be uneditable.

Reply:
What if we take the values retrieved from the DB and put them into the setFirstName() etc variables?

Reply:
I think that would work, and then you'd want to change the value of each TextField to firstName, lastName, etc. and not profile.firstName, profile.lastName, etc.